client/components/widget/query/query-result-data-service.js

```diff
@@ -226,26 +228,39 @@ QueryResultDataService.prototype.fetchResults = function(widget) {
         isSelected);
 
     promise.then(angular.bind(this, function(response) {
-      if (response.data.error) {
+      if (goog.isDefAndNotNull(response.data.error)) {
+        if (goog.string.isEmptySafe(response.data.error)) {
+          response.data.error = ERR_UNEXPECTED;
+        }
         this.errorService_.addError(ErrorTypes.DANGER, response.data.error);
         deferred.reject(response.data);
       } else {
         let rows = response.data.totalRows;
         let size = response.data.totalBytesProcessed;
         let time = response.data.elapsedTime;
-        let job = response.data.jobReference.jobId;
+        let jobReference = response.data.jobReference;
 
-        widget.state().datasource.job_id = job;
+        if (goog.isDefAndNotNull(jobReference)) {
+          widget.state().datasource.job_id = jobReference.jobId;
```

Comment:
Not at present; we just don't build the string if the data isn't there, and we do when it is.  This class is a prime example of still needing proper factoring and encapsulation for extensibility, but for now it suppresses errors.  And even if it's BigQuery, there are conditions where it won't return this string (we've had a few failures that occur before the jobid is returned).